python/ray/tests/test_projects.py

```diff
@@ -65,3 +69,51 @@ def test_project_no_validation():
     path = os.path.join(TEST_DIR, "project_files")
     with pytest.raises(subprocess.CalledProcessError):
         subprocess.check_call(["ray", "project", "validate"], cwd=path)
+
+
+@contextmanager
+def _chdir_and_back(d):
+    old_dir = os.getcwd()
+    try:
+        os.chdir(d)
+        yield
+    finally:
+        os.chdir(old_dir)
+
+
+def test_session_start_default_project():
+    test_dir = os.path.join(TEST_DIR,
+                            "project_files/session-tests/project-pass")
+
+    with _chdir_and_back(test_dir):
+        runner = CliRunner()
+        with patch.multiple(
+                "ray.projects.scripts",
+                create_or_update_cluster=DEFAULT,
+                rsync=DEFAULT,
+                exec_cluster=DEFAULT,
+        ) as mock_calls:
+            result = runner.invoke(start, [])
+            assert result.exit_code == 0
+
+    create_or_update_cluster_call = mock_calls["create_or_update_cluster"]
+    exec_cluster_call = mock_calls["exec_cluster"]
+
+    loaded_project = ray.projects.load_project(test_dir)
+
+    assert create_or_update_cluster_call.call_count == 1
+    _, kwargs = create_or_update_cluster_call.call_args
+    assert kwargs["config_file"] == loaded_project["cluster"]
+
+    commands_executed = []
+    for _, kwargs in exec_cluster_call.call_args_list:
+        commands_executed.append(kwargs["cmd"].replace(
```

Comment:
I'm not sure what this is doing :)

Reply:
It's adding `cmd` that were passed into autoscaler `exec` methods. I'll add a comment here. We are filtering out `cd {project_name}` because we are always `cd`ing into the project directory.